### ISSUE TITLE
Error incorrectly fires if multiple CTEs use --- separator

### ISSUE THREAD

**Issue by arthurbellis** [looker-open-source/look-at-me-sideways#43]
Thanks @fabio-looker for LAMS, my team at mercari.com has found it really useful to help us write maintainable SQL and LookML.

Ran into an interesting case that my team couldn't figure out until we debugged LAMS. :-)

We expected this to pass the linter:

```
  WITH
  
  step_1 AS (
    SELECT
      account_id AS pk1_account_id
      ---
    , COUNT(*)
    FROM users
    GROUP BY 1
  )
  
  , step_2 AS (
    SELECT
      pk2_tenant_id
    , pk2_user_id
      ---
    , MAX(start) as last_login
    FROM sessions
    GROUP BY 1,2
  )

  SELECT "foo" as bar
```

but got an error saying that the second CTE had mismatching primary key field counts:

> Primary Key columns in \" SELECT pk2_ten\"  in foo declare 2 column(s), but there are 1

It turned out this works if we use trailing commas, so that's a fine workaround:

```
  WITH
  
  step_1 AS (
    SELECT
      account_id AS pk1_account_id,
      ---
      COUNT(*)
    FROM users
    GROUP BY 1
  )
  
  , step_2 AS (
    SELECT
      pk2_tenant_id,
      pk2_user_id,
      ---
      MAX(start) as last_login
    FROM sessions
    GROUP BY 1,2
  )

  SELECT "foo" as bar
```

But the root cause seems to be that the rule parser is only replacing `---` once, so it misses the second one. I put together a quick diff w/ a test and will submit a pull request.

**Comment by fabio-looker** [comment]
Thank you so much! I'll review this later today

**Comment by fabio-looker** [comment]
Should be fixed in 1.4.2 now available on NPM

**Comment by arthurbellis** [comment]
Thanks @fabio-looker for the super-quick merge. ❤️
